### ISSUE TITLE
Normaliser les données

### ISSUE THREAD

**Issue by arnaudschaeffer** [LabexOBVIL/RCF-19#1]
@Virginie28 ,

Pour pouvoir importer le fichier RCF3_metadata, il faut dans un premier temps construire le schéma de la base de donnée et normaliser les données.

Fichier de base :
https://docs.google.com/spreadsheets/d/1T4pOR4E7XeqrVZF2nL9aoDHlVTzaYtgA5AH4uuhOjss/edit#gid=0

Pour cela, j'aurais besoin de bien comprendre la structure de ce fichier.

## Onglet Piece

* Titre : pas de problème
* Genre : est-ce qu'il s'agit d'un champs multi-valué ? Dans les valeurs, on peut trouver des valeurs avec un `/`. Est-ce que cela doit être considéré comme une valeur (`tragi-comédie / tragédie`) ou comme deux valeurs (`tragi-comédie` et `tragédie`) ?

Cela a une importance à plusieurs niveaux :

* en prenant l'option une valeur, si on recherche les pièces du genre `tragédie`, les pièces associés à `tragi-comédie / tragédie` ne ressortiront pas.
* en prenant l'option multi-valeué, si on recherche les pièces du genre `tragédie`, les pièces associés à `tragi-comédie` et `tragédie` ressortiront. On pourra également chercher les pièces ayant les deux genres `tragi-comédie` et `tragédie` .

* Catégorie : même relfexion ici. Est-ce que le champs doit être multi-valué ? Autre point, je pense que cette catégorie n'est pas bien normalisé et on a la même valeur écrit sous différente forme. Ex : `prose/vers`, `prose / vers` et `vers et prose`. Est-ce qu'il s'agit de la (des) même(s) catégorie(s) ?
Est-ce que tu peux normaliser cette colonne stp ?

* Date entrée CF et date création : il y a deux formats présents, mais je pense pouvoir les traiter via un script pour t'éviter de repasser sur le fichier de plus de 2000 lignes...

**Comment by arnaudschaeffer** [comment]
> Date entrée CF et date création : il y a deux formats présents, mais je pense pouvoir les traiter via un script pour t'éviter de repasser sur le fichier de plus de 2000 lignes...

J'ai pu traiter tous les formats de date du fichier. A savoir :

* YYYY-MM-DD : OK
* DD/MM/YYYY : OK
* YYYY-00-00 : remplacé par YYYY-01-01
* YYYY-MM : remplacé par YYYY-MM-01
* YYYY-MM- DD : suppression de l'espace indésirable

Est-ce que le remplacement des mois jours manquant par le premier jour du mois (et de l'année) te convient ? Sinon, que proposes tu ?

Les espaces dans la date posent problème.

Certaines pièces n'ont pas de nombres d'acte défini. Est-ce qu'il faut laisser cette valeur vide ? La remplacer par 1 ou 0 ?

**Comment by Virginie28** [comment]
Bonjour Arnaud,

Je suis désolée, j'ai fait une série de visios pour des cours ce matin. 
Il faut prendre les fichiers excel que je 'avais transmis l'après-midi où nous avons parlé : tout y est harmonisé !

**Comment by Virginie28** [comment]
Oui, c'est parfait comme solution ! Est-ce que tu travailles directement sur le drive ? 
Je vais normaliser les dates comme tu l'as fait sur les fichiers qui je viens d'ajouter comme ok (c'étaient les fichiers excel).

Pour répondre à tes questions plus haut:

Onglet Piece
Titre : pas de problème
Genre : est-ce qu'il s'agit d'un champs multi-valué ? oui 
Dans les valeurs, on peut trouver des valeurs avec un /. = deux valeurs 

Catégorie : même relfexion ici. Est-ce que le champs doit être multi-valué ? oui 

Autre point, je pense que cette catégorie n'est pas bien normalisé et on a la même valeur écrit sous différente forme. Ex : prose/vers, prose / vers et vers et prose. Est-ce qu'il s'agit de la (des) même(s) catégorie(s) ? Effectivement, cela concerne quelques entrées, je viens de corriger sur les fichiers ok. Le "et" doit en fait être remplacé par un /.
Il s'agit de plusieurs valeurs. 
Il y a trois types : 
-vers
-prose
-vers libres
et des combinaisons:
-vers / prose
-vers / vers libres
-prose / vers libres
J'ai normalisé dans 'Pièces ok".

**Comment by Virginie28** [comment]
En fait, je crois qu'il vaut mieux pas de date du tout qu'une date erronée.
Donc j'ai repris ça. Maintenant il faut utiliser les deux fichiers Auteurs ok et Pièces ok sur le drive : ils sont propres.

**Comment by arnaudschaeffer** [comment]
@Virginie28 , très bien !

Je vais récupérer le contenu du drive et le ré-intégrer. 

C'est bien noté pour les champs multi valués genre et catégories.
